### DISCUSSION TITLE
Type hint using Model.field

### DISCUSSION THREAD

**Discussion by UlfurOrn** [pydantic/pydantic#3252] [Feature Request]
Hi! I have an idea for a feature that could be added (Given that it is not already possible). Take the following code for instance:
```
class Human(BaseModel):
    id: int
    name: str
    children: List[Child]

class Child(BaseModel):
    id: int
    name: str

# This is a dictionary where the key is the ID of a human
# and the value is a list of the humans children's names.
dictionary: Dict[int, List[str]] = {}
```

What I would like to be able to do is the following:
```
class Human(BaseModel):
    id: int
    name: str
    children: List[Child]

class Child(BaseModel):
    id: int
    name: str


dictionary: Dict[Human.id, List[Child.name]] = {}
```
Essentially, what I wan't is for my editor to recognize that Human.id is of type int and to give editor help accordingly while increasing readability at the same time.

I'm not sure if this is even possible. But it would be a handy feature to have :)

**Comment by UlfurOrn** [comment]
When I think about it this is probably something that would need to be handled by the editor but I'll let the professionals decide on that

**Reply by dmontagu** [reply]
I think the syntax you want here is unlikely to be a thing we’ll support as annotations generally don’t become class attributes just by existing in python. It would be possible to support, but it would be challenging to make it work with type checkers/IDEs and isn’t super pythonic in general.

However, I think you may be interested in typing.NewType, which may be a good fit for your use case, you can do `HumanId = typing.NewType(‘HumanId’, str)` and then use that as the annotation for the id field of Human, and for the keys of your dict, and get many of the benefits I think you are looking to get.
